killer/killer_windows.py

```diff
@@ -60,14 +66,14 @@ def detect_tray(self):
     def detect_ethernet(self):
         # TODO: Add enum for:
         #   https://github.com/Lvl4Sword/Killer/wiki/Windows-Connection-Status-Codes
-        for x in wmi.WMI().Win32_NetworkAdapter():
-            if x.NetConnectionStatus is not None:
-                # This can contain quite a few things including Ethernet, Bluetooth, and Wireless
-                log.debug('%s %d %s', x.MacAddress, x.NetConnectionStatus, x.Name)
-
-                if x.MacAddress == self.config['windows']['ethernet_interface']:
-                    if x.NetConnectionStatus == 7:
-                        self.kill_the_system('Ethernet')
+        ipconfig_cmd = subprocess.check_output(['ipconfig', '/all']).decode()
+        for each in ipconfig_cmd.split('\r\n\r\n'):
+            mac_address = re.findall(MAC_ADDRESS_REGEX, each)
+            if mac_address == self.config['windows']['ethernet_interface']:
+                log.debug('MAC Address Detected: %s', mac_address)
+                media_state = re.findall(MEDIA_STATE_REGEX, each)
```

Comment:
You can probably put the findall() call in the if statement instead of a intermediate variable that isn't used

Reply:
the ``mac_address = re.findall(MAC_ADDRESS_REGEX, each)`` you mean? That's used in log.debug() though.